File: src/main/java/nodes/statements/MethodNode.java
package nodes;

import java.util.List;

public class MethodNode extends BaseNode{
    private String  accessModifier,type,name;
    private List<Parameter>parameters;
    private List<Statement>statements;

    public MethodNode(String accessModifier, String type, String name, List<Parameter> parameters, List<Statement> statements) {
        this.accessModifier = accessModifier;
        this.type = type;
        this.name = name;
        this.parameters = parameters;
        this.statements = statements;
    }

    @Override
    public String toString() {
        StringBuilder params = new StringBuilder();
        for (Parameter parameter:parameters)
            params.append(parameter.toString()).append(",");
        params.deleteCharAt(params.length() - 1);


        return accessModifier+" "+type+" "+name+"("+;
    }
}

File: src/main/kotlin/nodes/statements/MethodNode.kt
package nodes.statements

import nodes.BaseNode
import nodes.expressions.Parameter

class MethodNode(
    private val accessModifier: String,
    private val type: String,
    private val name: String,
    private val parameters: List<Parameter>,
    private val statements: List<Statement>
) : BaseNode() {
    override fun toString(): String {
        val params = StringBuilder()
        for (parameter in parameters) params.append(parameter.toString()).append(",")
        params.deleteCharAt(params.length - 1)
        return "$accessModifier $type $name("
    }
}